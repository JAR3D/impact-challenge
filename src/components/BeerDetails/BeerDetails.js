import classes from "./BeerDetails.module.css";

const BeerDetails = ({ name, description, labels }) => {
  return (
    <div className={classes.details}>
      <h2>{name}</h2>
      {labels ? (
        <img src={labels.medium} alt={name} />
      ) : (
        <p>No label available</p>
      )}
      <p>{description ? description : "No description available"}</p>
    </div>
  );
};

export default BeerDetails;
