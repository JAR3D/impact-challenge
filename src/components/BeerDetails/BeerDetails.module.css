.details {
  width: 50%;
  text-align: center;
  background-color: #ffdd7f;
}
