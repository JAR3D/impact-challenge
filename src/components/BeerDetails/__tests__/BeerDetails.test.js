import { render as renderTL } from "@testing-library/react";

import BeerDetails from "../BeerDetails";

describe("BeerDetails", () => {
  const render = (props) => renderTL(<BeerDetails {...props} />);

  it("should render", () => {
    const { baseElement } = render({
      name: "Foo",
      description: "Bar",
      labels: {
        medium:
          "https://brewerydb-images.s3.amazonaws.com/beer/UJGpVS/upload_K4w7gB-medium.png",
      },
    });
    expect(baseElement).toMatchSnapshot();
  });
});
