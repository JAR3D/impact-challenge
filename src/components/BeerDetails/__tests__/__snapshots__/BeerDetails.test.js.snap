// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerDetails should render 1`] = `
<body>
  <div>
    <div
      class="details"
    >
      <h2>
        Foo
      </h2>
      <img
        alt="Foo"
        src="https://brewerydb-images.s3.amazonaws.com/beer/UJGpVS/upload_K4w7gB-medium.png"
      />
      <p>
        Bar
      </p>
    </div>
  </div>
</body>
`;
